### PR TITLE
[SPARK-11292][SQL] Python API for text data source

### PR DESCRIPTION
**Pull request by rxin**
Adds DataFrameReader.text and DataFrameWriter.text.

### CODE REVIEW

**Comment by rxin** [comment]
cc @davies

File: python/pyspark/sql/readwriter.py [hunk 1]
```diff
@@ -193,10 +194,22 @@ def parquet(self, *paths):
         """
         return self._df(self._jreader.parquet(_to_seq(self._sqlContext._sc, paths)))
 
+    @ignore_unicode_prefix
+    @since(1.6)
+    def text(self, path):
```
Comment (davies) [inline python/pyspark/sql/readwriter.py]: Should we also support multiple files or directories, as for parquet?

Comment (rxin) [inline python/pyspark/sql/readwriter.py]: Could be a good idea. If we do that maybe we should do it for all data sources though, and maybe not part of this?

**Comment by yhuai** [comment]
merging to master.

**Pull request closed without merging**